### PR TITLE
Improvement to paper wallet

### PR DESCRIPTION
**Pull request by ngvtuan**
##### Commits
- [x] Improvement to paper wallet

##### FreedCamp
- [x] [Improvement to paper wallet](https://freedcamp.com/view/2956183/tasks/39534129)

### CODE REVIEW

File: src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.module.css [hunk 1]
```diff
@@ -34,3 +34,32 @@
...
+  display: block;
+  width: 36px;
+  height: 36px;
+  border-radius: 50%;
+  border: none;
+  background-color: transparent;
+}
+
+.btn_download:hover {
+  background-color: rgba(255, 255, 255, 0.1);
```
Comment (adrianobetkal) [inline src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.module.css:51]: For now it is okay, but mind that each color will be taken from `theme/colors` soon - we should not write any colors on the fly, but keep all the colors in one place instead

Comment (ngvtuan) [reply src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.module.css:51]: Ok @adrianobetkal

**Review by br4in3x — COMMENTED** [review]
Please resolve @adrianobetkal comments and think it's good to go

File: src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.tsx [hunk 2]
```diff
@@ -20,6 +21,154 @@ type PastelPaperWalletModalProps = {
   }
 }
 
+type PDFDocumentProps = {
```
[the remark this answers is not in the document]
Comment (ngvtuan) [reply src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.tsx]: I have updated the type here

File: src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.tsx [hunk 3]
```diff
@@ -20,6 +21,154 @@ type PastelPaperWalletModalProps = {
...
+  address: string
+  privateKey: string
+  currencyName: string
+  title: string
+}
+
+const splitStringIntoChunks = (s: string, numChunks: number) => {
```
[the remark this answers is not in the document]
Comment (ngvtuan) [reply src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.tsx]: I have updated the descriptive names

File: src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.tsx [hunk 4]
```diff
@@ -20,6 +21,154 @@ type PastelPaperWalletModalProps = {
...
+  if (numChunks > s.length) {
+    return [s]
+  }
+
+  if (s.length < 16) {
```
[the remark this answers is not in the document]
Comment (ngvtuan) [reply src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.tsx]: Updated the condition

File: src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.tsx [hunk 5]
```diff
@@ -20,6 +21,154 @@ type PastelPaperWalletModalProps = {
...
+    return [s]
+  }
+
+  const chunkSize = Math.round(s.length / numChunks)
+  const chunks = []
+
+  for (let i = 0; i < numChunks - 1; i++) {
```
[the remark this answers is not in the document]
Comment (ngvtuan) [reply src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.tsx]: I have updated

File: src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.tsx [hunk 6]
```diff
@@ -20,6 +21,154 @@ type PastelPaperWalletModalProps = {
...
+          <View style={pdfStyles.contentTop}>
+            <View style={pdfStyles.topMedia}>
+              <QRCodeGEnerator address={address} />
+            </View>
+            <View style={pdfStyles.contentWrapper}>
+              <Text style={pdfStyles.contentTitle}>
+                {currencyName} Address (Sapling)
+              </Text>
+              <Text style={pdfStyles.contentValue}>
+                {address.replace(/(.{40})/g, `$1${breakChar}`)}
```
[the remark this answers is not in the document]
Comment (ngvtuan) [reply src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.tsx]: Updated

File: src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.tsx [hunk 7]
```diff
@@ -32,79 +181,31 @@ export default function PastelPaperWalletModal({
...
-    },
-    topImg: {
-      width: '100%',
-    },
-    mainImg: {
-      width: '100%',
-    },
-  })
+  const generateFileName = () => {
+    const addr = splitStringIntoChunks(address, 6)[0]
```
[the remark this answers is not in the document]
Comment (ngvtuan) [reply src/features/pastelPaperWalletGenerator/PastelPaperWalletModal.tsx]: Updated

File: src/features/pastelPaperWalletGenerator/__test__/pastelPaperWalletModal.test.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
+import {
+  closePastelPaperWalletModal,
+  openPastelPaperWalletModal,
+  pastelPaperWalletModalReducer,
+} from '../PastelPaperWalletModalSlice'
+
+describe('PastelPaperWalletModal reducer', () => {
+  test('returns correct default state', () => {
+    // Act
+    const s = pastelPaperWalletModalReducer(undefined, { type: 'test' })
```
[the remark this answers is not in the document]
Comment (ngvtuan) [reply src/features/pastelPaperWalletGenerator/__test__/pastelPaperWalletModal.test.ts:10]: Updated

**Review by adrianobetkal — APPROVED** [review]

**Pull request merged**